api/app/db/crud/weather_models.py

```diff
@@ -241,6 +241,33 @@ def get_station_model_predictions(
     return query
 
 
+def get_latest_station_model_prediction_per_day(session: Session,
+                                                station_codes: List,
+                                                model: str,
+                                                start_date: datetime.datetime,
+                                                end_date: datetime.datetime):
+
+    query = session.query(WeatherStationModelPrediction, PredictionModelRunTimestamp, PredictionModel).\
+        filter(WeatherStationModelPrediction.station_code.in_(station_codes)).\
+        filter(WeatherStationModelPrediction.prediction_timestamp >= start_date).\
+        filter(WeatherStationModelPrediction.prediction_timestamp <= end_date).\
+        filter(PredictionModelRunTimestamp.id ==
+               WeatherStationModelPrediction.prediction_model_run_timestamp_id).\
+        filter(PredictionModelRunTimestamp.prediction_model_id == PredictionModel.id,
+               PredictionModel.abbreviation == model).\
+        order_by(WeatherStationModelPrediction.station_code).\
+        order_by(WeatherStationModelPrediction.prediction_timestamp).\
+        group_by(WeatherStationModelPrediction.station_code, func.date(WeatherStationModelPrediction.prediction_timestamp)).\
+        having(
+            WeatherStationModelPrediction.prediction_timestamp == session.query(
+                func.max(WeatherStationModelPrediction.prediction_timestamp))
+            .filter(func.date(WeatherStationModelPrediction.prediction_timestamp) == func.date(WeatherStationModelPrediction.prediction_timestamp))
+            .filter(WeatherStationModelPrediction.station_code == WeatherStationModelPrediction.station_code)
+            .subquery()
+            .c.prediction_timestamp)
+    return query
```

Comment:
Does this return a query that is ready to be executed? Or is the query already executed and this is a list of results/rows?

Reply:
It returns a query that can be iterated over to grab each row. I'm not sure how it works under the hood I think it does sort of cursor based iterator.